presto-hive/src/test/java/io/prestosql/plugin/hive/parquet/TestParquetDecimalScaling.java

```diff
@@ -83,13 +86,55 @@ public void testReadingMatchingPrecision(String tableName, boolean forceFixedLen
         dropTable(tableName);
     }
 
+    @Test(dataProvider = "different-rescalings")
+    public void testReadingRescaledDecimals(String tableName, boolean forceFixedLengthArray, int precision, int scale, List<String> values, int schemaPrecision, int schemaScale, List<String> expected)
+    {
+        createTable(tableName, schemaPrecision, schemaScale);
+
+        writeParquetDecimalsRecord(
+                getParquetWritePath(tableName),
+                ImmutableList.of(new ParquetDecimalInsert("value", forceFixedLengthArray, precision, scale, values)));
+
+        assertValues(tableName, schemaScale, expected);
+
+        dropTable(tableName);
+    }
+
+    @Test(dataProvider = "roundable-decimals")
+    public void testReadingRoundedDecimals(String tableName, boolean forceFixedLengthArray, int precision, int scale, List<String> values, int schemaPrecision, int schemaScale, List<String> expected)
```

Comment:
Would be good to add a javadoc to each method explaining what test covers, perhaps along with an example. Terms "rescaled", "rounded", "roundable" are not obvious.